src/api/SetTokenAPI.ts

```diff
@@ -152,6 +152,51 @@ export class SetTokenAPI {
     return quantity.mod(naturalUnit).eq(ZERO);
   }
 
+  /**
+   * Convenience function to calculate the units of components transferred given an issue quantity.
+   *
+   * @param  setAddress    Address of the Set
+   * @param  quantity      Quantity to be checked
+   * @return boolean       List of units
+   *
+   */
+  public async calculateComponentUnitsTransfered(setAddress: Address, quantity: BigNumber): Promise<BigNumber[]> {
+    const [naturalUnit, componentUnits] = await Promise.all([
+      this.setToken.naturalUnit(setAddress),
+      this.setToken.getUnits(setAddress),
+    ]);
+
+    return _.map(componentUnits, componentUnit => calculatePartialAmount(componentUnit, quantity, naturalUnit));
+  }
+
+  /**
+   * Convenience function to calculate the units of component transferred given an issue quantity.
+   *
+   * @param  setAddress        Address of the Set
+   * @param  componentAddress  Address of the component
+   * @param  quantity          Quantity to be checked
+   * @return boolean           Unit transfered in BigNumber format
+   *
+   */
+  public async calculateComponentUnitTransfered(
```

Comment:
the naming here makes me think I need to pass in the transaction id of a transfer request. what's the context of the transferring here?